src/userCLI/src/main/kotlin/org/imperial/mrc/hint/userCLI/App.kt

```diff
@@ -15,54 +16,49 @@ Usage:
     app user-exists <email>
 """
 
-fun main(args: Array<String>)
-{
+fun main(args: Array<String>) {
     val options = Docopt(doc).parse(args.toList())
     val addUser = options["add-user"] as Boolean
     val removeUser = options["remove-user"] as Boolean
     val userExists = options["user-exists"] as Boolean
+    val props = DatabaseProperties()
+    val dataSource = DbConfig().dataSource(props.url, props.user, props.password)
 
-    //Start a background HintApplication so we can get an ApplicationContext in order to get Autowired user repo
-    val applicationContext = SpringApplication.run(HintApplication::class.java)
-
-    try
-    {
-        val userCLI = UserCLI(applicationContext)
-        val result = when
-        {
+    try {
+        val userCLI = UserCLI(dataSource)
+        val result = when {
             addUser -> userCLI.addUser(options)
             removeUser -> userCLI.removeUser(options)
             userExists -> userCLI.userExists(options)
             else -> ""
         }
 
         println(result)
-    }
-    catch(e: Exception)
-    {
+    } catch (e: Exception) {
         System.err.println(e.message)
         exitProcess(1)
     }
     finally {
-        applicationContext.close()
+        dataSource.connection.close()
     }
 }
 
-class UserCLI(private val appContext: ApplicationContext)
-{
-    fun addUser(options: Map<String, Any>, userRepository: UserRepository=userRepository()): String
-    {
+class UserCLI(dataSource: DataSource) {
+
+    private val profileService = DbProfileService(dataSource, SecurePasswordEncoder())
```

Comment:
I suppose we could use a HintDbProfileService here, rather than the base class, though at the moment that wouldn't make any practical difference as the only override relates to login validation.